doc/whats_new/v1.3.rst

```diff
@@ -137,6 +137,16 @@ Changelog
 - |Enhancement| All selectors in :mod:`sklearn.feature_selection` will preserve
   a DataFrame's dtype when transformed. :pr:`25102` by `Thomas Fan`_.
 
+:mod:`sklearn`
+..............
+
+- |Feature| Added a new option `skip_parameter_validation`, to the function
```

Comment:
I don't it makes the validation framework public. Instead it makes the public informed that there's validation going on when they call the public API (which is something everybody expects I believe), but without giving details on how it's done.

For now the frameworks is still not fully stable and we still have to fix a few things and change the API from time to time so I don't think it's ready to be made fully public. I'd rather wait for it to be a bit more stable, maybe when all public functions will be validated as well.

We can still document for third party devs that there's validation, it can be costly, here's how mitigate that.

Reply:
> We can still document for third party devs that there's validation

I think that this would be a good start. This is still fine to have a "private" decorator. This could be done of course in another PR.